docs/tutorials/time-ranges.md

```diff
@@ -0,0 +1,127 @@
+# Time Ranges
+
+Overview
+----------
+
+A Timeline and all of the Tracks and Stacks with in it work together to place
+the Clips, Gaps and Transitions relative to each other in time. You can think
+of this as a 1-dimensional coordinate system. Simple cases of assembling Clips
+into a Track will lay the Clips end-to-end, but more complex cases involve
+nesting, cross-dissolves, trimming, and speed-up/slow-down effects which can
+lead to confusion. In an attempt to make this easy to work with OpenTimelineIO 
+uses the following terminology and API for dealing with time ranges.
+
+Note: You probably also want to refer to [Timeline Structure](otio-timeline-structure.html).
+
+Clips
+-------
+
+There are several ranges you might want from a Clip.
+
+- Ranges within the Clip and its media:
+
+    - `clip.available_range()`
+
+    The `available_range()` method on Clip returns a TimeRange that tells you
+    how much media is available via the Clip's `media_reference`. If a Clip 
+    points to a movie file on disk, then this should tell you how long that 
+    movie is and what timecode it starts at. For example: "wedding.mov" starts 
+    at timecode 01:00:00:00 and is 30 minutes long.
+
+    Note that `available_range()` may return `None` if the range is not known.
+
+    - `clip.source_range`
+
+    Setting the `source_range` property on a Clip will trim the Clip to only 
+    that range of media.
+
+    Note that `source_range` may be `None` indicating that the Clip should use
+    the full `available_range()` whatever that may be. If both `source_range`
+    and `available_range()` are None, then the Clip is invalid. You need at
+    least one of those.
+
+    Usually this will be a shorter segment than the `available_range()` but this
+    is not a hard constraint. Some use cases will intentionally ask for
+    a Clip that is longer (or starts earlier) than the available media as a way
+    to request that new media (a newly animated take, or newly recorded audio) 
+    be made to fill the requested `source_range`.
```

Comment:
You may want to note that `source_range` is in the coordinate frame of the `available_range`.  In other words, if the `available_range` is from frame 5 to 20, and the source range is from 0 to 10, that doesn't mean frame 5-15 in the available range, but rather frames 0 to 10, of which only 5 are in the available range (which may indicate a problem).

Reply:
Do we define what that coordinate frame is anywhere?  Or how it relates to the coordinate frame of the media reference?  Regardless, sounds good.

Reply:
That stuff is also just looking ahead to when we start back in on the time stuff.